rest/rest-smartproxy/src/main/java/org/ow2/proactive_grid_cloud_portal/smartproxy/RestSmartProxyImpl.java

```diff
@@ -462,89 +462,89 @@ public void registerAsListener() throws NotConnectedException, PermissionExcepti
-        return ((ISchedulerClient) _getScheduler()).isTaskFinished(jobId, taskName);
+        return _getScheduler().isTaskFinished(jobId, taskName);
     }
 
     @Override
     public TaskResult waitForTask(String jobId, String taskName, long timeout) throws UnknownJobException,
             NotConnectedException, PermissionException, UnknownTaskException, TimeoutException {
-        return ((ISchedulerClient) _getScheduler()).waitForTask(jobId, taskName, timeout);
+        return _getScheduler().waitForTask(jobId, taskName, timeout);
     }
 
     @Override
     public List<JobResult> waitForAllJobs(List<String> jobIds, long timeout)
             throws NotConnectedException, UnknownJobException, PermissionException, TimeoutException {
-        return ((ISchedulerClient) _getScheduler()).waitForAllJobs(jobIds, timeout);
+        return _getScheduler().waitForAllJobs(jobIds, timeout);
     }
 
     @Override
     public Map.Entry<String, JobResult> waitForAnyJob(List<String> jobIds, long timeout)
             throws NotConnectedException, UnknownJobException, PermissionException, TimeoutException {
-        return ((ISchedulerClient) _getScheduler()).waitForAnyJob(jobIds, timeout);
+        return _getScheduler().waitForAnyJob(jobIds, timeout);
     }
 
     @Override
     public Map.Entry<String, TaskResult> waitForAnyTask(String jobId, List<String> taskNames, long timeout)
             throws UnknownJobException, NotConnectedException, PermissionException, UnknownTaskException,
             TimeoutException {
-        return ((ISchedulerClient) _getScheduler()).waitForAnyTask(jobId, taskNames, timeout);
+        return _getScheduler().waitForAnyTask(jobId, taskNames, timeout);
     }
 
     @Override
     public List<Map.Entry<String, TaskResult>> waitForAllTasks(String jobId, List<String> taskNames, long timeout)
             throws UnknownJobException, NotConnectedException, PermissionException, UnknownTaskException,
             TimeoutException {
-        return ((ISchedulerClient) _getScheduler()).waitForAllTasks(jobId, taskNames, timeout);
+        return _getScheduler().waitForAllTasks(jobId, taskNames, timeout);
     }
 
     @Override
     public boolean pushFile(String spacename, String pathname, String filename, String file)
             throws NotConnectedException, PermissionException {
-        return ((ISchedulerClient) _getScheduler()).pushFile(spacename, pathname, filename, file);
+        return _getScheduler().pushFile(spacename, pathname, filename, file);
     }
 
     @Override
     public void pullFile(String space, String pathname, String outputFile)
             throws NotConnectedException, PermissionException {
-        ((ISchedulerClient) _getScheduler()).pullFile(space, pathname, outputFile);
+        _getScheduler().pullFile(space, pathname, outputFile);
     }
 
     @Override
     public boolean deleteFile(String space, String pathname) throws NotConnectedException, PermissionException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)